### PR TITLE
feat(navbar): create a navbar

### LINKED ISSUE #19: Navabr
**Task title**
Create a functional Navbar

**Task description**
Create a functional navbar, that cpntains:
- Logo
- Home Button
- Products Button
- Order Page
- Profile Button
- Search Button
ensure responsiveness as well

**Subtasks**
- [ ] Create Search Bar
- [x] Create Profile Button
- [x] Populate the Navbar

**Screenshots**
image

**Link to the component on Figma**
[Figma Link
](https://www.figma.com/file/FYZIsFvg1rLu50UnriUN8Y/Student-Store-Project?type=design&node-id=0-1&mode=design&t=0DdQfVRHQ85SzzLb-0)
[Additional Resource](https://github.com/202306-NEA-DZ-FEW/e-commerce-project-magnimarket/blob/main/src/components/Navbar/Navbar.jsx)

### PR DESCRIPTION
**Pull request by 0m3ga13**
in this changes we created a navbar using headless ui transition and disclosure, and we made sure it functionality in the mobile view

Resolve #19 

**Screenshots**
image
image
image

# Related Issue

- Resolve #19

### CODE REVIEW

File: src/components/navbar/Navbar.jsx [hunk 1]
```diff
@@ -1,5 +1,87 @@
...
+import { BiMenu, BiMenuAltRight, BiUser } from "react-icons/bi";
 
-export default Navbar;
+import MobileSidebar from "../mobileSidebar/MobileSidebar";
+import Portal from "../portal/Portal";
+import Searchbar from "../search/Searchbar";
+
+export default function Navbar() {
+    const navigation = [
+        { name: "Dashboard", href: "/" },
```
Comment (Bolphunga) [inline src/components/navbar/Navbar.jsx]: Change the "Dashnoard" to 'Home."

File: src/components/portal/Portal.jsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+"use client";
+
+import { createPortal } from "react-dom";
+
+const Portal = ({ children }) => {
+    if (typeof window === "object") {
+        const mount = document.body;
+        const el = document.createElement("div");
+
+        // useEffect(() => {
```
Comment (Bolphunga) [inline src/components/portal/Portal.jsx]: This commented section is unnecessary. It needs to be removed.

**Review by Bolphunga — CHANGES_REQUESTED** [review]

File: src/components/navbar/Navbar.jsx (hunk 1 above)
Comment (0m3ga13) [reply src/components/navbar/Navbar.jsx]: done

**Review by Bolphunga — APPROVED** [review]
Good job. Approved.

**Review by samiraTbl — APPROVED** [review]
done!

**Review by medshk — APPROVED** [review]
PR permitted

**Pull request merged**